### PR TITLE
Improve osd components repo cleanup logics

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Improve osd components repo cleanup logics

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-build/issues/4350

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
```

 info copying assets from `public/assets` to build
 info copying server source into the build and converting with babel
 info running yarn to install dependencies
 info compressing plugin into [searchRelevanceDashboards-2.12.0.zip]
 info cleaning up compression temporary artifacts
Done in 15.34s.
+ cd /tmp/tmpyx9aja4b/searchRelevanceDashboards
+ echo 'COPY searchRelevanceDashboards.zip'
COPY searchRelevanceDashboards.zip
+ cp -r ../OpenSearch-Dashboards/plugins/searchRelevanceDashboards/build/searchRelevanceDashboards-2.12.0.zip builds/plugins/
+ cleanup_all
+ File_Delete ../OpenSearch-Dashboards/plugins/searchRelevanceDashboards
+ FILE_NAME=../OpenSearch-Dashboards/plugins/searchRelevanceDashboards
+ echo Attempt to rm ../OpenSearch-Dashboards/plugins/searchRelevanceDashboards
Attempt to rm ../OpenSearch-Dashboards/plugins/searchRelevanceDashboards
+ '[' -z ../OpenSearch-Dashboards/plugins/searchRelevanceDashboards ']'
+ echo 'Removing ../OpenSearch-Dashboards/plugins/searchRelevanceDashboards'
Removing ../OpenSearch-Dashboards/plugins/searchRelevanceDashboards
+ rm -rf -- ../OpenSearch-Dashboards/plugins/searchRelevanceDashboards
2024-01-17 23:27:57 INFO     Appended searchRelevanceDashboards component in build manifest.
2024-01-17 23:27:57 INFO     Recording plugins artifact for searchRelevanceDashboards: plugins/searchRelevanceDashboards-2.12.0.zip (from /tmp/tmpyx9aja4b/searchRelevanceDashboards/builds/plugins/searchRelevanceDashboards-2.12.0.zip)
2024-01-17 23:27:57 INFO     Checked /tmp/tmpyx9aja4b/searchRelevanceDashboards/builds/plugins/searchRelevanceDashboards-2.12.0.zip (2.12.0.0)
2024-01-17 23:27:57 INFO     Successfully built searchRelevanceDashboards
2024-01-17 23:27:57 INFO     Created build manifest /home/ci-runner/opensearch-build/tar/builds/opensearch-dashboards/manifest.yml
2024-01-17 23:27:57 INFO     Keeping /tmp/tmpyx9aja4b
2024-01-17 23:27:57 ERROR    Failed plugins are ['customImportMapDashboards', 'indexManagementDashboards', 'notificationsDashboards', 'alertingDashboards', 'securityAnalyticsDashboards', 'securityDashboards']
2024-01-17 23:27:57 INFO     Done.
```

File: scripts/components/ganttChartDashboards/build.sh [hunk 1]
```diff
@@ -10,8 +10,12 @@
 set -ex
 
 # vars / libs
-. ../../../lib/shell/file_management.sh
-PLUGIN_PATH=$PWD
+SCRIPT_DIR=`dirname $(realpath $0)`
+. $SCRIPT_DIR/../../../lib/shell/file_management.sh
+# For hybrid plugin it actually resides in 'dashboards-visualizations/gantt-chart'
```
Comment (prudhvigodithi) [inline scripts/components/ganttChartDashboards/build.sh:15]: @peterzhuamazon does this work for windows?

Comment (peterzhuamazon) [reply scripts/components/ganttChartDashboards/build.sh:15]: I think so, it is using the bash.exe anyway during the run.
Tho I do not have a server to test for now.
If needed, I can do a verification tomorrow once I obtain a server. Thanks.

**Review by prudhvigodithi — APPROVED** [review]

File: scripts/components/ganttChartDashboards/build.sh (hunk 1 above)
Comment (peterzhuamazon) [reply scripts/components/ganttChartDashboards/build.sh:15]: After talking to @prudhvigodithi we will use the jenkins pipeline to test it.

**Pull request merged**